sprint01/created-basic-apis-for-practice

AfaqAhmed-XGrid opened the pull request:
Addresses #2

mafaqahmed committed:
sprint01/created-basic-apis-for-practice

junaidtariqxgrid commented on index.js:
```diff
@@ -0,0 +1,31 @@
+
+app.get('/', (req, res) => {
+    res.status(200);
+    res.json({message: 'Welcome to the root URL of the server'})
```
Please add strings to constants

FahadJalal04 replied:
Not addressed

junaidtariqxgrid commented on index.js:
```diff
@@ -0,0 +1,31 @@
+    res.json({message: 'Welcome to the root URL of the server'})
+})
+
+app.get('/hello', (req, res) => {
```
Please add these single quoted strings to constants

FahadJalal04 replied:
Same

AfaqAhmed-XGrid replied:
Done!

junaidtariqxgrid commented on package.json:
```diff
@@ -6,15 +6,10 @@
   "scripts": {
```
All URLs and strings should be added to constants.js

AfaqAhmed-XGrid replied:
Done!

FahadJalal04 commented on index.js:
```diff
@@ -0,0 +1,31 @@
+const express = require('express');
+const app = express();
+const path = require('path');
```
Categories import statements and add a comment.
e.g. 
// Module Imports
All imports related to npm modules

// Component Imports
....

AfaqAhmed-XGrid replied:
Done!

FahadJalal04 commented on index.js:
```diff
@@ -0,0 +1,31 @@
+
+app.get('/', (req, res) => {
+    res.status(200);
+    res.json({message: 'Welcome to the root URL of the server'})
```
Move your controller functions to a separate file and add middleware for authentication and authorization.

AfaqAhmed-XGrid replied:
Done!

FahadJalal04 commented on index.js:
```diff
@@ -0,0 +1,31 @@
+
+app.get('/hello', (req, res) => {
+    res.set('Content-Type', 'text/html')
+    res.send('<h1>Welcome to the hello URL of the server</h1>')
```
The response should be a JSON object like: 
{
   success: 0/1,
   message: 'your message',
   data: {}
}

AfaqAhmed-XGrid replied:
Done!

mafaqahmed committed:
resolved all conflicts

categorized imports, imrproved fodler structure, restrained pkg.json to change package version, rectified response structure, created constant.js file

FahadJalal04 commented on modules/practice/practice.controller.js:
```diff
@@ -0,0 +1,19 @@
+//constant imports
+const constants = require("./practice.constant");
+
+const printMessage = (req, res) => {
```
This function should be an async function

AfaqAhmed-XGrid replied:
I did not use await keywork so why to make it an async function?

FahadJalal04 replied:
Commented for future ref.

AfaqAhmed-XGrid replied:
OKay!

FahadJalal04 commented on modules/practice/practice.controller.js:
```diff
@@ -0,0 +1,19 @@
+//constant imports
+const constants = require("./practice.constant");
+
```
Add function level comments

AfaqAhmed-XGrid replied:
Done!

FahadJalal04 commented on modules/practice/practice.controller.js:
```diff
@@ -0,0 +1,19 @@
+const constants = require("./practice.constant");
+
+const printMessage = (req, res) => {
+    res.status(200);
```
Use 2 spaces for indent

AfaqAhmed-XGrid replied:
Done!

FahadJalal04 commented on modules/practice/practice.controller.js:
```diff
@@ -0,0 +1,19 @@
+}
+
+const printName = (req, res) => {
+    const {name} = req.body
```
No need to extract the name from the body. You can directly use req.body.name

AfaqAhmed-XGrid replied:
Done!

FahadJalal04 commented on modules/practice/practice.route.js:
```diff
@@ -0,0 +1,19 @@
+//Package Imports
+const express  =require("express");
```
Why 2 spaces before the equal sign and no space after equal?

AfaqAhmed-XGrid replied:
Done!

FahadJalal04 commented on modules/practice/practice.route.js:
```diff
@@ -0,0 +1,19 @@
+
+router.get('/hello', createHTML)
+
+router.post('/api/post', printName)
```
Use a separate file for routes as discussed

AfaqAhmed-XGrid replied:
I have created all routes in a separate practice.route.js file

mafaqahmed committed:
added function comments and some other minor changes

FahadJalal04 reviewed (commented):
Replied

mafaqahmed committed:
implemented eslint and improved function comments
